plugins/tiledb/io/TileDBWriter.cpp

```diff
@@ -0,0 +1,345 @@
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<char>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Signed16:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<short>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Signed32:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<int>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Signed64:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<long>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Unsigned8:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<unsigned char>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Unsigned16:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<unsigned short>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Unsigned32:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<unsigned int>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Unsigned64:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<unsigned long>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::None:
+            default:
+                throw pdal_error("Unsupported attribute type for " + kv.first);
+
+        }
+    }
+
+    m_query->set_coordinates(coords);
+    m_query->submit();
+}
+
+void TileDBWriter::done(PointTableRef table)
+{
+    m_query->finalize();
+    m_array->close();
+
+    // write pipeline metadata sidecar
+    MetadataNode anon;
+    MetadataNode meta("pipeline");
+    anon.addList(meta);
+    // set output to tiledb reader
+    meta.add("type", "readers.tiledb");
+    meta.add("array_name", m_arrayName);
+    if (!getSpatialReference().empty() && table.spatialReferenceUnique())
+    {
+        // The point view takes on the spatial reference of that stage, if it had one.
+        meta.add("spatialreference", pdal::Utils::toString(getSpatialReference()));
+    }
+
+    meta.add("bounds", pdal::Utils::toString(m_bbox));
+
+    // serialize pipeline
+    std::ostream* ostr = FileUtils::createFile(m_arrayName + ".json");
+    pdal::Utils::toJSON(anon, *ostr);
```

Comment:
Discussed saving this with VFS to support all the backends of TileDB.

Reply:
:+1: